Source/Controllers/DataControllers/SPTableData.m

```diff
@@ -516,9 +541,87 @@ - (NSDictionary *) informationForTable:(NSString *)tableName fromDatabase:(NSStr
     
     if (changeEncoding) [mySQLConnection restoreStoredEncoding];
 
+    SPLog(@"cols count: %lu", ((NSDictionary*)[tableData safeObjectForKey:@"columns"]).count);
+
     return tableData;
 }
 
+- (NSArray *)createTableSyntaxFromView:(NSString *)tableName withSyntaxResult:(NSArray *)syntaxResult {
+
+    SPLog(@"createTableSyntaxFromView");
+
+    NSString *queryStr = [NSString stringWithFormat:@"SHOW COLUMNS FROM %@", [tableName backtickQuotedString]];
+
+    SPLog(@"queryStr: %@", queryStr);
+
+    SPMySQLResult *theResult = [mySQLConnection queryString:queryStr];
+
+    NSMutableString *viewCreateStr = [[NSMutableString alloc] init];
+
+    [viewCreateStr appendFormat:@"CREATE TEMPORARY TABLE `%@` (\n", tableName];
```

Comment:
Will this require special permissions to the database in order to do? In other words, can I create a temporary table even from a readonly connection?